### PR TITLE
Refactor CDI documentation

### PR DESCRIPTION
**Pull request by slaskawi**
https://issues.jboss.org/browse/ISPN-7039

### CODE REVIEW

File: documentation/src/main/asciidoc/user_guide/cdi.adoc [hunk 1]
```diff
@@ -1,36 +1,47 @@
 ==  CDI Support
-Infinispan includes integration with link:http://docs.oracle.com/javaee/6/tutorial/doc/giwhb.html[Contexts and Dependency Injection (better known as CDI)]
-via Infinispan's `infinispan-cdi` module.
-Configuration and injection of Infinispan's Cache interface is provided, and it is planned to bridge Cache listeners to the CDI event system as well.
-The module also provide partial support of the JCache (JSR-107) caching annotations - for further details see link:$$https://docs.google.com/document/d/1YZ-lrH6nW871Vd9Z34Og_EqbX_kxxJi55UrSn4yL2Ak/edit?hl=en&amp;pli=1#heading=h.jdfazu3s6oly$$[Chapter 8] of the JCACHE specification.
+Infinispan includes integration with link:http://www.cdi-spec.org[Contexts and Dependency Injection (better known as CDI)] via Infinispan's `infinispan-cdi-embedded` or `infinispan-cdi-remote` module. 
+CDI is part of link:http://www.oracle.com/technetwork/java/javaee/tech/index-jsp-142185.html[JEE specification] and aims for managing beans' lifecycle inside the container.
```
Comment (tristantarrant) [inline documentation/src/main/asciidoc/user_guide/cdi.adoc]: There is no such thing as JEE. It's Java EE

File: documentation/src/main/asciidoc/user_guide/cdi.adoc [hunk 2]
```diff
@@ -117,16 +132,18 @@ public class Config {
 
 ----
 
-It's also possible to override the default embedded cache manager used.
-The new default cache manager produced must have the @Default qualifier and the scope @ApplicationScoped . 
+It's also possible to override the default `EmbeddedCacheManager`.
+Newly created manager must have default qualifiers and Application scope.
```
Comment (tristantarrant) [inline documentation/src/main/asciidoc/user_guide/cdi.adoc]: s/Newly created/The newly created/

File: documentation/src/main/asciidoc/user_guide/integrations.adoc [hunk 3]
```diff
@@ -1227,6 +1227,12 @@ $$3.$$ Configure persistence.xml to use Infinispan instead of JBoss Cache:
 
 ----
 
+===  Using Infinispan CDI integration
+
+Infinispan supports both injecting remote and embedded Caches as well as bridging events at Cache and CacheManager level.
+
+For more information, have a look at our link:../user_guide/user_guide.html#_cdi_support[CDI paragraph].
```
Comment (tristantarrant) [inline documentation/src/main/asciidoc/user_guide/integrations.adoc]: the underscore prefix on paragraph anchors is a relic of the past. Also since this is an anchor in this same document, you can use link:#cdi_support[CDI paragraph].

File: documentation/src/main/asciidoc/user_guide/cdi.adoc (hunk 1 above)
Comment (slaskawi) [inline documentation/src/main/asciidoc/user_guide/cdi.adoc]: Fixed

File: documentation/src/main/asciidoc/user_guide/cdi.adoc (hunk 2 above)
Comment (slaskawi) [inline documentation/src/main/asciidoc/user_guide/cdi.adoc]: Fixed

File: documentation/src/main/asciidoc/user_guide/cdi.adoc [hunk 4]
```diff
@@ -263,50 +290,68 @@ public class Config {
...
 .Producer method scope
-NOTE: To work properly the producers must have the scope @ApplicationScoped . Otherwise each injection of cache will be associated to a new instance of cache manager. 
+NOTE: To work properly the producers must have the scope @ApplicationScoped . Otherwise each injection of cache will be associated to a new instance of cache manager.
 
 === Use JCache caching annotations
 
 TIP: There is now a separate module for JSR 107 (JCACHE) integration, including API.  See <<_using_infinispan_as_a_jsr107_jcache_provider, this chapter>> for details.
 
-The infinispan-cdi module provides a partial support of JCache caching annotations.
+The `infinispan-jcache` and `infinispan-jcache-remote` modules provide partial support of JCache caching annotations with CDI.
```
Comment (slaskawi) [inline documentation/src/main/asciidoc/user_guide/cdi.adoc]: There need to be both artifacts (JCache and CDI) on the classpath.. let me rephrase it..

File: documentation/src/main/asciidoc/user_guide/integrations.adoc (hunk 3 above)
Comment (slaskawi) [inline documentation/src/main/asciidoc/user_guide/integrations.adoc]: ok, thanks for the hint!

**Comment by slaskawi** [comment]
@tristantarrant updated and addressed all comments.

**Comment by tristantarrant** [comment]
Pushed, thanks
